eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -34,32 +44,41 @@ class ContentHandler extends AbstractInMemoryPersistenceHandler implements Conte
 
     protected function init(): void
     {
-        $this->getContentInfoTags = function (ContentInfo $info, array $tags = []) {
-            $tags[] = 'content-' . $info->id;
+        $tagGenerator = $this->tagGenerator;
+
+        $this->getContentInfoTags = function (ContentInfo $info, array $tags = []) use ($tagGenerator) {
+            $tags[] = $tagGenerator->generate(self::CONTENT_TAG, [$info->id]);
 
             if ($info->mainLocationId) {
                 $locations = $this->persistenceHandler->locationHandler()->loadLocationsByContent($info->id);
                 foreach ($locations as $location) {
-                    $tags[] = 'location-' . $location->id;
+                    $tags[] = $tagGenerator->generate(self::LOCATION_TAG, [$location->id]);
                     foreach (explode('/', trim($location->pathString, '/')) as $pathId) {
-                        $tags[] = 'location-path-' . $pathId;
+                        $tags[] = $tagGenerator->generate(self::LOCATION_PATH_TAG, [$pathId]);
                     }
                 }
             }
 
             return $tags;
         };
-        $this->getContentInfoKeys = function (ContentInfo $info) {
+        $this->getContentInfoKeys = function (ContentInfo $info) use ($tagGenerator) {
             return [
-                'ez-content-info-' . $info->id,
-                'ez-content-info-byRemoteId-' . $this->escapeForCacheKey($info->remoteId),
+                $tagGenerator->generate(self::CONTENT_INFO_TAG, [$info->id], true),
+                $tagGenerator->generate(
+                    self::CONTENT_INFO_BY_REMOTE_ID_TAG,
+                    [$this->escapeForCacheKey($info->remoteId)],
+                    true
+                ),
```

Comment:
I've just realized that we are using Tag Generator for both, generating tags and generating cache item keys what doesn't feel right from architectual point of view (even if from implementation might be that same concept). 

In this scenario we should to split key/tags generation in one of the possible ways:

a) method level (rename TagGenerator to e.g. CacheIdentityGenerator and add generateKey method with own pool of patterns)
b) class level (introduce KeyGenerator class)
c) service level

Reply:
I like the idea, but can this be done as a follow-up to unblock this PR perhaps?

Reply:
We can do it as followup. Let's not change a logic in this PR, main focus was to shorten tags.

Reply:
@konradoboza please add it to the list, so this followup can come-up as soon as possible.